clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -717,6 +717,29 @@ public void testLowestPriorityChannel() throws Exception {
         assertNull(selector.lowestPriorityChannel());
     }
 
+    @Test
+    public void testMetricsCleanupOnSelectorClose() throws Exception {
+        Metrics metrics = new Metrics();
+        RuntimeException runtimeException = new RuntimeException();
+        Selector selector = new ImmediatelyConnectingSelector(5000, metrics, time, "MetricGroup", channelBuilder, new LogContext()) {
+            @Override
+            public void close(String id) {
+                throw runtimeException;
+            }
+        };
+        assertTrue(metrics.metrics().size() > 1);
+        String id = "0";
+        selector.connect(id, new InetSocketAddress("localhost", server.port), BUFFER_SIZE, BUFFER_SIZE);
+        try {
+            selector.close();
+        } catch (Exception e) {
+            assertEquals(e, runtimeException);
```

Comment:
You should use `assertThrows` instead. Otherwise we need a `fail` after `close()` to make sure we actually raise an exception.

Reply:
Fixed.